### PR TITLE
Fixed invalid version number and merge upstream v1.12.1

### PR DESCRIPTION
**Pull request by gushil**
Closes #
- Fixed pyxform cannot installed using pip / requirements file in Form Designer because of invalid version number according to https://peps.python.org/pep-0440/
- Merging latest upstream release of pyxform

#### What are the regression risks?
No. All tests are passed.

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `nosetests` and verified all tests pass
- [x] run `black pyxform tests` to format code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ===============
-pyxform v1.0.3-oc
+pyxform v1.0.3+oc
```
Comment (svadla-oc) [inline README.rst:2]: why is "-" not valid?

Comment (gushil) [reply README.rst:2]: In addition to that (1.0.3-oc) version is rejected by pip install, it probably not complies to
- https://peps.python.org/pep-0440/#public-version-identifiers
- https://peps.python.org/pep-0440/#local-version-identifiers

We can see the summary of possible version number here:
- https://peps.python.org/pep-0440/#examples-of-compliant-version-schemes

Thanks.

Comment (svadla-oc) [reply README.rst:2]: got it, thanks. In that case, I think it would be better to go with something like v1.0.3.oc instead of v1.0.3+oc.

Comment (gushil) [reply README.rst]: Unfortunately, 1.0.3.oc is not accepted.
image

Comment (gushil) [reply README.rst]: Public version should complyto: `[N!]N(.N)*[{a|b|rc}N][.postN][.devN]`
Local version should comply to: `<public version identifier>[+<local version label>]`

Should we go with 1.0.3+oc?

Thanks.

Comment (svadla-oc) [reply README.rst]: @gushil yes, please. Thanks for trying the other option.

Comment (gushil) [reply README.rst:2]: @svadla-oc

Done.

**Review by svadla-oc — APPROVED** [review]
